docs/root/intro/arch_overview/advanced/matching/matching_listener.rst

```diff
@@ -0,0 +1,63 @@
+.. _arch_overview_matching_listener:
+
+Matching Filter Chains in Listeners
+===================================
+
+Envoy listeners implement the :ref:`matching API <envoy_v3_api_msg_.xds.type.matcher.v3.Matcher>` for selecting a filter
+chain based on a collection of :ref:`network inputs <extension_category_envoy.matching.network.input>`. Matching is done
+once per connection. Connections are drained when the associated named filter chain configuration changes, but not when
+the filter chain matcher is the only updated field in a listener.
+
+The action in the matcher API must be a string value corresponding to the name of the filter chain. If there is no
+filter chain with the given name, the match fails, and the :ref:`default filter chain
+<envoy_v3_api_field_config.listener.v3.Listener.default_filter_chain>` is used if specified, or the connection is
+rejected. Filter chain matcher requires that all filter chains in a listener are uniquely named.
+
+The matcher API replaces the existing filter :ref:`filter_chain_match
+<envoy_v3_api_field_config.listener.v3.FilterChain.filter_chain_match>` field. When using the matcher API, the filter
+chain match field is ignored and should not be set.
+
+Examples
+########
+
+Detect TLS traffic
+******************
+
+The following examples uses :ref:`tls_inspector <config_listener_filters_tls_inspector>` listener filter to detect
+whether the transport appears to be TLS, in which case the matcher in the listener selects the filter chain ``tls``.
+Otherwise, the filter chain ``plaintext`` is used.
+
+.. literalinclude:: _include/listener_tls.yaml
+    :language: yaml
+    :linenos:
+    :lines: 37-56
```

Comment:
if we are going to include `linenos` and the sample doesnt start from the first line,  then i think we also need to set `lineno-start` to the first line

it would also be good to include a caption so users can get the entire file:

```
:caption: :download:`listener_tls.yaml <_include/listener_tls.yaml>`
```

Reply:
same below

Reply:
Done. I removed the line numbers, don't seem essential.

Reply:
i think they are a helpful guide to the user - both visually that there is a full file that the snippet is taken from and for quickly finding the snippet in the config

i wont block on it tho as usage is inconsistent elsewhere